docs/build/guides/conventions/upgrading-contracts.mdx

```diff
@@ -0,0 +1,333 @@
+// because the old contracts Wasm is expected to exist in storage.
+let contract_id = env.register_contract_wasm(None, old_contract::Wasm);
+
+let client = old_contract::Client::new(&env, &contract_id);
+let admin = Address::random(&env);
+client.init(&admin);
+
+assert_eq!(1, client.version());
+```
+
+We install the new contract's Wasm:
+
+```rust
+let new_wasm_hash = install_new_wasm(&env);
+```
+
+Then we run the upgrade, and verify that the upgrade worked:
+
+```rust
+client.upgrade(&new_wasm_hash);
+assert_eq!(2, client.version());
+```
+
+## Build the contract
+
+To build the contract `.wasm` files, run `stellar contract build` in both `upgradeable_contract/old_contract` and `upgradeable_contract/new_contract` in that order.
+
+Both `.wasm` files should be found in both contract `target` directories after building both contracts:
+
+```
+target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_old_contract.wasm
+```
+
+```
+target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_new_contract.wasm
+```
+
+## Run the contract
+
+If you have [`stellar-cli`] installed, you can invoke contract functions. Deploy the old contract and install the Wasm for the new contract.
+
+Navigate to to `upgradeable_contract/old_contract`
+
+```sh
+stellar contract deploy \
+  --wasm target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_old_contract.wasm  --source alice  \
+  --network testnet
+```
+
+- When you deploy a smart contract to a network, you need to specify an identity that will be used to sign the transactions. Change the `alice` to your own [identity].
+
+[identity]: ../../smart-contracts/getting-started/setup#configure-an-identity
+
+You should see this similar contract ID after you have ran the deploy command:
+
+```
+CAS6FKBXGVXFGU2SPPPJJOIULJNPMPR6NVKWLOQP24SZJPMB76TGH7Y3
+```
+
+Navigate to `upgradeable_contract/new_contract` and run the following command:
+
+```
+stellar contract install \
+--source-account alice \
+--wasm target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_new_contract.wasm \
+--rpc-url https://soroban-testnet.stellar.org:443 \--network-passphrase "Test SDF Network ; September 2015" \
+--network testnet
+```
+
+You should see this Wasm hash from the install command:
+
+```
+aa24c81289997ad815489b29db337b53f284cca5aba86e9a8ae5cef7d31842c2
+```
+
+You also need to call the `init` method so the `admin` is set. This requires us to setup some [identities].
+
+[identities]: ../../smart-contracts/getting-started/setup.mdx#configure-an-identity
+
+Given that we have previously setup an identity, to get the identity address, we run the following command:
+
+```sh
+stellar keys address alice
+```
+
+Example output:
+
+```
+GCJ2R5ST4UQP2D4F54Y3IIAQKPMLMEEZCNZ3PEDKY4AGDYEMYUC2MOO7
+```
+
+Navigate to `upgradeable_contract/old_contract` and now call `init` with this key (make sure to substitute with the key you generated):
```

Comment:
I don't _think_ the "Navigate to ..." step is necessary, and we can just call `init`, since it's already deployed to the network, and we're calling it using the `--id CAS6...` argument. That said, it's not gonna hurt anything to be in the directory. So, I'm fine with it either in or out.